### ISSUE TITLE
BUD Build fails: error on commit:: "btrfs driver does not support mount options"

### ISSUE THREAD

**Issue by SwitchedToGitlab** [containers/buildah#1825]
**KIND: BUG**

**Description**
buildah bud fails to commit due to BTRFS related issue, [most likely related to this issue.](https://github.com/containers/storage/pull/416) Podman has a [nearly identical bug, listed here.](https://github.com/containers/libpod/issues/3773) 

If at all possible, it would be super helpful to know which package the BTRFS driver is in so I can downgrade it and see if the issue persists, and until a fix can work it's way through the process. I noticed that Buildah hasn't been upgraded since this stopped working on this machine, so I'm assuming that the issue is in a supporting driver package that may have been recently updated. Thanks! 

**Steps to reproduce the issue:**
1. ```sudo buildah bud --tag opensuse-leap-custom .```
2. Wait a while
3. Commit step of the  build fails with output below.

**Describe the results you received:**
```
STEP 5: COMMIT opensuse-leap-custom
error committing container for step {Env:[ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin] Command:run Args:[zypper --non-interactive install vim git curl wget zsh;] Flags:[] Attrs:map[] Message:RUN zypper --non-interactive install vim git curl wget zsh; Original:RUN zypper --non-interactive install vim git curl wget zsh;}: error copying layers and metadata for container "7b097f830d4dd202a6c8857e4295db9e9d167d353ecb41e0d7970752aedbde5e": Error initializing source containers-storage:leap-working-container: error extracting layer "741c6b76642ee8b603a91cf7f47a8c19c51a3763ba5d9dbf934b05d293416668": btrfs driver does not support mount options
```

**Describe the results you expected:**

Build to succeed, world domination to commence.

**Output of `rpm -q buildah` or `apt list buildah`:**
```
buildah-1.10.1-1.1.x86_64
```

**Output of `buildah version`:**
```
buildah version 1.10.1 (image-spec 1.0.1, runtime-spec 1.0.1-dev)
```

**Output of `cat /etc/*release`:**

```
NAME="openSUSE Tumbleweed"
# VERSION="20190824"
ID="opensuse-tumbleweed"
ID_LIKE="opensuse suse"
VERSION_ID="20190824"
PRETTY_NAME="openSUSE Tumbleweed"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:tumbleweed:20190824"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
LOGO="distributor-logo"
```

**Output of `uname -a`:**

```
Linux rocinante 5.2.9-1-default #1 SMP Fri Aug 16 20:25:11 UTC 2019 (80c0ffe) x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`: (most comments removed)**

```
[storage]

# Default Storage Driver
driver = "btrfs"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# Storage options to be passed to underlying storage drivers

# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# If specified, use OSTree to deduplicate files with the overlay backend
ostree_repo = ""


[storage.options.thinpool]
```

**Comment by rhatdan** [comment]
Comment out the storage options in /etc/containers/storage.conf and this will probably work.

**Comment by vrothberg** [comment]
@rhatdan, maybe we should just ignore (and log) those options instead of errorring out? In any case, I'd like the error string to point users into the right direction (e.g., pointing to the used `storage.conf`).

**Comment by rhatdan** [comment]
We would have to modify containers/storage for this.  But I am fine with making these a warning rather then an error.  The only issue would be if these settings were required for security.
A better solution would be to make them driver specific.

IE setting overlay.storage_options rather then just storage options.  Since these default options don't necessarily apply to other drivers.

**Comment by vrothberg** [comment]
I like the idea of making the driver specific

**Comment by rhatdan** [comment]
Could you open an issue on containers/storage to make storage options driver specific.

**Comment by vrothberg** [comment]
> Could you open an issue on containers/storage to make storage options driver specific.

https://github.com/containers/storage/issues/423

**Comment by SwitchedToGitlab** [comment]
Sorry folks but commenting the storage options out of ```/etc/containers/storage.conf``` does not resolve the issue. I have tried just the storage options sections of both ```/etc/containers/storage.conf``` and ```~/.config/containers/storage.conf``` (which was installed by my distro fyi) as well as commenting out the entire file in both locations and I'm still getting the same error output as listed above.

**Comment by rhatdan** [comment]
Is podman working?

**Comment by rhatdan** [comment]
Ok this is a bug in buildah, it was fixed in containers/storage after buildah was released.

https://github.com/containers/storage/commit/e3f1fe0cb355fed87eb6e03bd6c873f470067e83

We need to release a new buildah version @TomSweeneyRedHat

This is fixed in master branch

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Time to release V1.11.0

**Comment by TomSweeneyRedHat** [comment]
Closing as this was fixed in Bulidah v1.11.0 due to vendoring: https://github.com/containers/storage/pull/416
